### PR TITLE
Check index unique & skip prefetch for non-heap relation

### PR DESCRIPTION
**Pull request by gfphoenix78**
* Check index unique in the AM callback if set. The callback is designed to support unique index for AO tables, but it could also work for other table AM.
* Skip prefetch page for non-heap relation. Prefetch loads a typical pg buffer to scan, but it can't work for non-page based storage.

fix #ISSUE_Number

---

### Change logs

_Describe your change clearly, including what problem is being solved or what feature is being added._

_If it has some breaking backward or forward compatibility, please clary._

### Why are the changes needed?

_Describe why the changes are necessary._

### Does this PR introduce any user-facing change?

_If yes, please clarify the previous behavior and the change this PR proposes._

### How was this patch tested?

_Please detail how the changes were tested, including manual tests and any relevant unit or integration tests._

### Contributor's Checklist

Here are some reminders and checklists before/when submitting your pull request, please check them:

- [ ] Make sure your Pull Request has a clear title and commit message. You can take [git-commit](https://github.com/cloudberrydb/cloudberrydb/blob/main/.gitmessage) template as a reference.
- [ ] Sign the Contributor License Agreement as prompted for your first-time contribution(*One-time setup*).
- [ ] Learn the [coding contribution guide](https://cloudberrydb.org/contribute/code), including our code conventions, workflow and more.
- [ ] List your communication in the [GitHub Issues](https://github.com/cloudberrydb/cloudberrydb/issues) or [Discussions](https://github.com/orgs/cloudberrydb/discussions) (if has or needed).
- [ ] Document changes.
- [ ] Add tests for the change
- [ ] Pass `make installcheck`
- [ ] Pass `make -C src/test installcheck-cbdb-parallel`
- [ ] Feel free to request `cloudberrydb/dev` team for review and approval when your PR is ready🥳

### CODE REVIEW

File: src/backend/access/table/tableam.c [hunk 1]
```diff
@@ -288,7 +288,7 @@ table_index_fetch_tuple_check(Relation rel,
 	 * overhead is significant for AO/CO relations. For details, please refer to
 	 * table_index_unique_check().
 	 */
-	if (RelationIsAppendOptimized(rel))
+	if (rel->rd_tableam->index_unique_check)
```
Comment (my-ship-it) [inline src/backend/access/table/tableam.c:291]: Update comments?

Comment (gfphoenix78) [reply src/backend/access/table/tableam.c:291]: Updated. The current comment is good enough about the changes. The code change applies for AO/CO and other non-heap tables(including pax) that per-tuple overhead is expensive.

File: src/backend/executor/nodeBitmapHeapscan.c [hunk 2]
```diff
@@ -73,6 +73,20 @@ static inline void BitmapPrefetch(BitmapHeapScanState *node,
...
+ * the prefetch will not work for them if they have
+ * non-standard page-based storage.
+ */
+static inline bool
+RelationSupportPrefetch(Relation rel)
+{
+	bool non_standard;
+	non_standard = (rel->rd_rel->relkind == RELKIND_RELATION ||
+					rel->rd_rel->relkind == RELKIND_MATVIEW) &&
+					!RelationIsHeap(rel);
```
Comment (avamingli) [inline src/backend/executor/nodeBitmapHeapscan.c:87]: non_standard will be false here if relkind is not RELKIND_RELATION, that support prefresh.Is it right?

Comment (gfphoenix78) [reply src/backend/executor/nodeBitmapHeapscan.c:87]: Bitmap scan will not scan other type of relkind, right?

Comment (avamingli) [reply src/backend/executor/nodeBitmapHeapscan.c:87]: Then it seems we don't need to judege relkind here.

**Review by my-ship-it — APPROVED** [review]
LGTM

**Review by avamingli — APPROVED** [review]

File: src/backend/executor/nodeBitmapHeapscan.c (hunk 2 above)
Comment (gfphoenix78) [reply src/backend/executor/nodeBitmapHeapscan.c]: Has removed. Thanks for your suggestion.

**Review by avamingli — APPROVED** [review]

**Pull request merged**